homeassistant/components/zwave_js/cover.py

```diff
@@ -304,6 +332,79 @@ def __init__(
+    """Representation of a Z-Wave Window Covering cover device."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, driver: Driver, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize."""
+        super().__init__(config_entry, driver, info)
+        pos_value: ZwaveValue | None = None
+        tilt_value: ZwaveValue | None = None
+
+        # If primary value is for position, we have to search for a tilt value
+        if info.primary_value.property_key in COVER_POSITION_PROPERTY_KEYS:
+            pos_value = info.primary_value
+            tilt_value = next(
+                (
+                    value
+                    for property_key in COVER_TILT_PROPERTY_KEYS
+                    if (
+                        value := self.get_zwave_value(
+                            CURRENT_VALUE_PROPERTY, value_property_key=property_key
+                        )
+                    )
+                ),
+                None,
+            )
+        # If primary value is for tilt, there is no position value
+        else:
+            tilt_value = info.primary_value
+
+        # Set position and tilt values if they exist. If the corresponding value is of
+        # the type No Position, we remove the corresponding set position feature.
+        for set_values_func, value, set_position_feature in (
+            (self._set_position_values, pos_value, CoverEntityFeature.SET_POSITION),
+            (self._set_tilt_values, tilt_value, CoverEntityFeature.SET_TILT_POSITION),
+        ):
+            if value:
+                set_values_func(
+                    value,
+                    stop_value=self.get_zwave_value(
+                        WINDOW_COVERING_OPEN_PROPERTY,
+                        value_property_key=value.property_key,
+                    ),
+                )
+                if value.property_key in NO_POSITION_PROPERTY_KEYS:
+                    assert self._attr_supported_features
+                    self._attr_supported_features ^= set_position_feature
+
+        additional_info: list[str] = []
+        for value in (self._current_position_value, self._current_tilt_value):
+            if value and value.property_key_name:
+                additional_info.append(
+                    value.property_key_name.removesuffix(f" {NO_POSITION_SUFFIX}")
+                )
+        self._attr_name = self.generate_name(additional_info=additional_info)
+        self._attr_device_class = CoverDeviceClass.WINDOW
+
+    @property
+    def _fully_open_tilt(self) -> int:
+        """Return position to open cover tilt."""
+        return SlatStates.OPEN
+
+    @property
+    def _fully_closed_tilt(self) -> int:
+        """Return position to close cover tilt."""
+        return SlatStates.CLOSED_1
+
+    @property
+    def _tilt_range(self) -> int:
+        """Return range of valid tilt positions."""
+        return abs(SlatStates.CLOSED_2 - SlatStates.CLOSED_1)
```

Comment:
Isn't `SlatStates` an enum so the properties here will be constant? We shouldn't use a property if it's a constant. Just store a constant at the module level. Properties are for dynamically calculated values.

Reply:
can I override a property function with a class attribute?

Reply:
Oh, I see that it's dynamic by default in the mixin but here's it's a constant. Does it need to be fully dynamic in the mixin or can we use an attribute instead?

We don't need to change this now. It's ok to keep them properties.

Reply:
It may not need to be. Will look at that later 👍